hack/tests/ci-prepare.sh

```diff
@@ -30,6 +30,31 @@ stop_monitoring() {
     fi
 }
 
+cleanup() {
+    set +e
+
+    if [[ -n "$ARTIFACTS" ]]; then
+        exec &>"$ARTIFACTS/cleanup"
+    fi
+
+    if [[ -n "$MONITORING" ]]; then
+      stop_monitoring
+    fi
+
+    make artifacts
+
+    if [[ -n "$T" ]]; then
```

Comment:
the reason we hadn't condensed all the cleanup()s into one so far is that they're subtly different.  This PR doesn't yet handle things neatly enough to make me want to condense them.  Example: I don't want to rely on a variable called $T in a library cleanup().  Possibly renaming $T and making a wrapper function that cleans up $T then calls cleanup()?  Or still trapping the `rm -rf "$T"` in the other script?  Needs more experimentation.

Reply:
Taking a look at all the cleanup functions they are mostly the same except for some of them having stop_monitoring, which as Angus points out is already idempotent: won't fail if monitoring wasn't started.
I do agree about T, I was just trying to maintain the status quo but that does look off to me, let me see how to best tackle that